pkg/controller/backup_controller.go

```diff
@@ -670,6 +670,16 @@ func (c *backupController) runBackup(backup *pkgbackup.Request) error {
 		if err != nil {
 			backupLog.Errorf("fail to wait VolumeSnapshot change to Ready: %s", err.Error())
 		}
+
+		// checkVolumeSnapshotReadyToUse could run for a long time. Update VolumeSnapshot array after it finishes.
+		err = c.kbClient.List(context.Background(), vsList, &kbclient.ListOptions{LabelSelector: selector})
+		if err != nil {
+			backupLog.Error(err)
+		}
+		if len(vsList.Items) >= 0 {
```

Comment:
No need to check the len(x) >= 0 and we should put it in the `else` clause
Same issue in line 666.